src/T.php

```diff
@@ -69,4 +71,9 @@ public static function string(): StringType
     {
         return new StringType();
     }
+
+    public static function nullable(Type $type): NullableType
```

Comment:
I'm not sure if `$type` should be typehinted. There is some inconsistency - `Collection` constructor doesn't typehint `$type` (as opposed to `Tuple`) and also `ValidatesType::resolveType` can work just with any callable (not only `Type`).

Reply:
You can keep the typehint here, I'm still thinking about whether one or the other would be better. I liked the idea of using a closure as a type on the fly, but it makes things a little more complicated. I'll probably change everything to always be a `Type`.